controllers/registry_controller.go

```diff
@@ -21,13 +21,20 @@ import (
 	"fmt"
```

Comment:
> Edits the way file transfers work for easier reuse of FTP clients.

this may warrant it's own commit.

Reply:
the changes are tightly coupled, not sure I can split this